test/integration/users.test.js

```diff
@@ -0,0 +1,156 @@
+const supertest = require('supertest');
+const { factory } = require('factory-girl');
+
+const dbHelper = require('../helpers/db');
+const usersTestData = require('../data/users');
+const app = require('../../app');
+const { errorMessages } = require('../../app/constants');
+
+require('../setup');
+
+const postUsersPath = '/users';
+const defaultHeaders = ['Accept', 'application/json'];
+
+describe('Create users POST  /users', () => {
+  describe('Should work correctly with satisfactory case', () => {
+    let response = {};
+
+    beforeEach(async done => {
```

Comment:
Is better to use a **beforeAll** here in order to do a one request to de endpoint for all cases tests. With beforeEach you will call the endpoint for each test.

Reply:
If I use a beforeAll, the database would still be truncated afterEach test, this would make fail the test that checks if the user was successfully stored in the database because there wouldn't be any user in DB

Reply:
you're right, ignore this comment :)